docs/src/main/asciidoc/security-oidc-auth0-tutorial.adoc

```diff
@@ -0,0 +1,743 @@
+[id="security-oidc-auth0-tutorial"]
+= Protect Quarkus web application by using `Auth0` OpenID Connect provider
+include::_attributes.adoc[]
+:diataxis-type: tutorial
+:categories: security,web
+
+xref:security-architecture.adoc[Quarkus Security] provides comprehensive OpenId Connect (OIDC) and OAuth2 support with its `quarkus-oidc` extension, supporting both xref:security-oidc-code-flow-authentication.adoc[Authorization code flow] and xref:security-oidc-bearer-token-authentication.adoc[Bearer token] authentication mechanisms.
+
+With Quarkus, you can easily configure OIDC providers such as link:https://www.keycloak.org/documentation[Keycloak], link:https://developer.okta.com/[Okta], link:https://auth0.com/docs/[Auth0], and other xref:security-openid-connect-providers.adoc[well-known social OIDC and OAuth2 providers].
+
+Learn how to use the Quarkus OpenID Connect extension (`quarkus-oidc`) together with https://auth0.com/docs/[Auth0] OIDC provider to protect your API endpoints.
+
+== Create Auth0 application
+
+Start with creating an `Auth0` application in the `Auth0` dashboard, for example, lets create a `QuarkusAuth0` web application:
+
+image::auth0-create-application.png[Create Auth0 application]
```

Comment:
@sberyozkin - Just some general comments about the screencaps instructional content.

- Screencaps are high maintenance. We typically try to avoid them for both accessibility and localization reasons.
- If the UIs in those screencaps get updated regularly and the docs don't get updated, we risk losing the trust of your reader or creating mismatch confusion.
- I would suggest you use them as judiciously as possible and ensure alt text is unique and descriptive to support visually impaired users.

Reply:
@sberyozkin - In saying that, _personally as a user/reader, I love them!_

Reply:
Hi @michelle-purcell I'm not exactly sure I understand, when you say `screencaps`, do you mean the images ? The chance of the text in them changing is very low, if it does it won't change a lot, I've been hoping to  make sure the new Auth0 users who may not be well familiar with the Auth0 setup steps, just follow this tutorial.
We also hope this tutorial be reviewed externally, once published. Ideally, it will be very easy for the reviewers to follow.

Does it clarify why I've added (more than usual) number of images ?

Reply:
I guess, since you liked it yourself, we can resolve this one :-) ?

Reply:
@michelle-purcell Let me resolve just to optimize the review space but please re-open if I've missed something

Reply:
@sberyozkin - Thanks for explaining the rationale here. 

Yes, I mean the screen capture images of the Auth0 UI and other UIs, especially the UI that we don't own. 

If you think they are definitely required and the maintenance risks are low then keep them. 

Important: The screen captures of the UI must be supplemental only for the novice user. So the key question is:
If you removed any of the UI images, would the user have enough information with the text alone to follow and complete the required actions? 

If the answer is no, then we need another iteration to enhance the textual instructions for accessibility support.

For every image in the guide, we also need to ensure that every image has an Alt text attribute that clearly describes the picture for someone who cannot see it. I can do the ALT text review of images after this initial iteration when the step text edits/comments/suggestions are closed out.
Make sense? Thanks.